### ISSUE TITLE
False positive?

### ISSUE THREAD

**Issue by paxbe** [validator/validator#510]
Nu Html Checker

Error: No p element in scope but a p end tag seen.

From line 617, column 6; to line 617, column 9

This error is reproduce-able, but i think my code is correct. See below for the source code:

    <p>
    	This is the part of the <abbr title="Document type Definition">DtD</abbr> string, that TidyHTML looks for, to detect the <abbr title="Document type Definition">DtD</abbr> in
    	the document.
    	For example,
    	<code>
    		HTML 4.01 Transitional
    	</code>
    	will be found in the detected <abbr title="Document type Definition">DtD</abbr> string:
    	<br>
    	<blockquote>
    		<code>
    			&lt;!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd"&gt;
    		</code>
    	</blockquote>
    </p>

**Comment by LeeKowalkowski** [comment]
Smaller document to reproduce:

```
<!DOCTYPE html>
<html>
<head><title>test</title></head>
<body><p><blockquote></blockquote></p></body>
</html>
```

I think putting a block-level element inside an element that does not allow them could have a better error message, however, the problem with the HTML is because [blockquotes are not permitted inside paragraphs](https://www.w3.org/TR/html5/dom.html#phrasing-content-1), it is assumed that the [paragraph's end tag has been omitted](https://www.w3.org/TR/html5/grouping-content.html#the-p-element):

> A p element's end tag may be omitted if the p element is immediately followed by an address, article, aside, blockquote, div, dl, fieldset, footer, form, h1, h2, h3, h4, h5, h6, header, hgroup, hr, main, nav, ol, p, pre, section, table, or ul, element, or if there is no more content in the parent element and the parent element is not an a element.

The [q element](https://www.w3.org/TR/html5/text-level-semantics.html#the-q-element) would work in place of blockquotes within paragraphs in your document.

**Comment by sideshowbarker** [comment]
While I agree the error message for this case would ideally be better, the current message is correct per the parsing algorithm in the HTML spec. This is just the way that HTML parsing works, and rewriting the message handling for this in the HTML parser code to not make clear to users what’s actually happening is arguably not an improvement. But if somebody wants to take a shot at writing up a patch to emit a friendlier message for this case I’m happy to review it.
